DomainContent/AvatarStore/CheckoutZone/checkoutZone.js

```diff
@@ -0,0 +1,255 @@
+//
+//  checkoutZone.js
+//
+//  Created by Rebecca Stankus on 9/29/17.
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+//  This zone will provide an area at which a user may purchase an item. When the avatar enters the zone wearing a 
+//  marketplace item, the item will appear as a small overlay. Scanning the overlay will cause the 
+//  the tablet to open to the marketplace home page for that item, allowing the user to quickly make the purchase.
+(function () {
+    var SHARED = Script.require('./attachmentZoneShared.js');
+    var MARKET_PLACE_ITEM_URL_PREFIX = 'https://metaverse.highfidelity.com/marketplace/items/';
+    var ITEM_HEIGHT = 0.1;
+    var OVERLAY_PREFIX = 'MP';
+    var TRANSFORMS_SETTINGS = 'io.highfidelity.avatarStore.checkOut.tranforms';
+    
+    var _this = this;
+    var isInZone = false;
+    var tableProperties, tableHeight, tableLength, tableID, spawnZ, spawnY, spawnX;
+    var zoneID;
+    var replicaList = [];
+    var replicaStoredTransforms = {};
+    var left = true;
+
+    this.preload = function(entityID) {
+        zoneID = entityID;
+    };
+
+    var getTransformForMarketplaceItems = function() {
+        return Settings.getValue(TRANSFORMS_SETTINGS, {});
+    };
+    
+    var getTransformsForMarketplaceItem = function(marketplaceID) {
+        var transformItems = getTransformForMarketplaceItems();
+        if (transformItems[marketplaceID] === undefined) {
+            return {
+                certificateTransforms: {},
+                unsortedTransforms: [],
+                lastUsedUnsortedTransformIndex: -1
+            };
+        }
+        return transformItems[marketplaceID];
+    };
+    
+    var addTransformForMarketplaceItem = function(marketplaceID, certificateID, transform) {
+        if (marketplaceID === undefined) {
+            return;
+        }
+        var marketplaceItemTransforms = getTransformForMarketplaceItems();
+        var marketplaceItemTransform = getTransformsForMarketplaceItem(marketplaceID);
+        if (certificateID !== undefined) {
+            marketplaceItemTransform.certificateTransforms[certificateID] = transform;
+        } else {
+            marketplaceItemTransform.unsortedTransforms.push(transform);
+        }
+        marketplaceItemTransforms[marketplaceID] = marketplaceItemTransform;
+        Settings.setValue(TRANSFORMS_SETTINGS, marketplaceItemTransforms);
+    };
+
+    // Get info on checkout stand so we can place copies of items on it for purchasing
+    // Find the position of the top of the stand at one end
+    var getCheckoutStandPosition = (function(){
+        var zoneChildren = Entities.getChildrenIDs(zoneID);
+        zoneChildren.forEach(function (childID) {
+            var name = Entities.getEntityProperties(childID, 'name').name;
```

Comment:
Is it more effective to retrieve the properties once and reuse them in the if statement? How often will the zone have children that aren't checkout tables?